File: screen-switcher-sample/src/main/java/com/jaynewstrom/screenswitchersample/first/FirstDialogFactory.java
package com.jaynewstrom.screenswitchersample.first;

import android.app.Dialog;
import android.content.Context;

import com.jnewstrom.screenswitcher.dialoghub.DialogFactory;

import javax.inject.Inject;

final class FirstDialogFactory implements DialogFactory {
    @Inject FirstDialogFactory() {
    }

    @Override public Dialog createDialog(Context context) {
        return new FirstDialog(context);
    }
}

File: sample-screen-first/src/main/java/com/jaynewstrom/screenswitchersample/first/FirstDialogFactory.kt
package com.jaynewstrom.screenswitchersample.first

import android.app.Dialog
import android.content.Context
import com.jaynewstrom.screenswitcher.dialogmanager.DialogFactory
import javax.inject.Inject

internal class FirstDialogFactory @Inject constructor() : DialogFactory {
    override fun createDialog(context: Context): Dialog {
        return FirstDialog(context)
    }
}

private class FirstDialog(context: Context) : Dialog(context) {
    init {
        setContentView(R.layout.first_dialog)
    }
}
